### PR TITLE
Import declarations should not accept visibility modifiers.

### LINKED ISSUE #359: Import declarations should not accept visibility modifiers
Currently the compiler accepts `public import Foo;`

We should report this as an error.

### PR DESCRIPTION
**Pull request by Kuinox**
Fix #359.

### CODE REVIEW

File: src/Draco.Compiler/Internal/Syntax/SyntaxErrors.cs [hunk 1]
```diff
@@ -135,4 +135,13 @@ internal static class SyntaxErrors
         severity: DiagnosticSeverity.Error,
         format: "illegal language element {0} in context",
         code: Code(14));
+
+    /// <summary>
+    /// There is a visibility modifier before an import statement.
+    /// </summary>
+    public static readonly DiagnosticTemplate UnexpectedVisibilityModifierBeforeImport = DiagnosticTemplate.Create(
```
Comment (Kuinox) [inline src/Draco.Compiler/Internal/Syntax/SyntaxErrors.cs:142]: Is that overkill ?

File: src/Draco.Compiler.Tests/Syntax/ParserTests.cs [hunk 2]
```diff
@@ -1802,4 +1806,15 @@ public void TestEmptyModuleDeclarationStatement()
...
+
+    [Fact]
+    public void TestImportDeclarationWithVisibilityModifier()
+    {
+        this.ParseDeclaration("public import Foo;");
+
+        this.N<UnexpectedDeclarationSyntax>();
+        {
+            this.InvalidT(TokenKind.KeywordPublic);
+        }
```
Comment (Kuinox) [inline src/Draco.Compiler.Tests/Syntax/ParserTests.cs:1818]: There is no helper for parsing multiple declaration, so I didn't tested that the rests of the tokens are present.

File: src/Draco.Compiler/Internal/Syntax/Parser.cs [hunk 3]
```diff
@@ -261,6 +261,13 @@ private DeclarationSyntax ParseDeclaration(DeclarationContext context)
...
         {
         case TokenKind.KeywordImport:
+            if (modifier is not null)
+            {
+                var info = DiagnosticInfo.Create(SyntaxErrors.UnexpectedVisibilityModifierBeforeImport, formatArgs: "declaration");
+                var diag = new SyntaxDiagnosticInfo(info, Offset: 0, Width: modifier.Width);
+                this.AddDiagnostic(modifier, diag);
+                return new UnexpectedDeclarationSyntax(modifier, SyntaxList.Create(Array.Empty<SyntaxNode>()));
+            }
             return this.ParseImportDeclaration();
```
Comment (LPeter1997) [inline src/Draco.Compiler/Internal/Syntax/Parser.cs:271]: A low hanging fruit would be to store the invalid token right in the import statement so the shape would be
```
ImportDeclaration
{
    IllegalToken(public),
    rest of the node...
}
```
Better tree shape overall.

File: src/Draco.Compiler/Internal/Syntax/SyntaxErrors.cs (hunk 1 above)
Comment (LPeter1997) [reply src/Draco.Compiler/Internal/Syntax/SyntaxErrors.cs:142]: It's totally fine

File: src/Draco.Compiler/Internal/Syntax/Parser.cs (hunk 3 above)
Comment (Kuinox) [reply src/Draco.Compiler/Internal/Syntax/Parser.cs:271]: The purpose was that the import would still be entirely valid and be correctly parsed.

**Review by LPeter1997 — APPROVED** [review]

**Pull request merged**